packages/test/test-end-to-end-tests/src/test/idCompressor.spec.ts

```diff
@@ -0,0 +1,326 @@
+import { SharedCell } from "@fluidframework/cell";
+import { IIdCompressor, SessionSpaceCompressedId } from "@fluidframework/runtime-definitions";
+import { SharedObjectCore } from "@fluidframework/shared-object-base";
+
+const mapId = "mapKey";
+const cellId = "cellKey";
+const testContainerConfig: ITestContainerConfig = {
+	registry: [
+		[mapId, SharedMap.getFactory()],
+		[cellId, SharedCell.getFactory()],
+	],
+	runtimeOptions: {
+		enableRuntimeIdCompressor: true,
+	},
+	fluidDataObjectType: DataObjectFactoryType.Test,
+};
+
+function getIdCompressor(dds: SharedObjectCore): IIdCompressor {
+	return (dds as any).runtime.idCompressor as IIdCompressor;
+}
+
+describeNoCompat("Runtime IdCompressor", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	beforeEach(() => {
+		provider = getTestObjectProvider();
+	});
+
+	let sharedMap1: SharedMap;
+	let sharedCell1: SharedCell;
+	let sharedMap2: SharedMap;
+	let sharedMap3: SharedMap;
+
+	beforeEach(async () => {
+		const container1 = await provider.makeTestContainer(testContainerConfig);
+		const dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+		sharedMap1 = await dataObject1.getSharedObject<SharedMap>(mapId);
+		sharedCell1 = await dataObject1.getSharedObject<SharedCell>(cellId);
+
+		const container2 = await provider.loadTestContainer(testContainerConfig);
+		const dataObject2 = await requestFluidObject<ITestFluidObject>(container2, "default");
+		sharedMap2 = await dataObject2.getSharedObject<SharedMap>(mapId);
+
+		const container3 = await provider.loadTestContainer(testContainerConfig);
+		const dataObject3 = await requestFluidObject<ITestFluidObject>(container3, "default");
+		sharedMap3 = await dataObject3.getSharedObject<SharedMap>(mapId);
+
+		sharedMap1.set("testKey1", "testValue");
+
+		await provider.ensureSynchronized();
+	});
+
+	it("has no compressor if not enabled", async () => {
+		provider.reset();
+		const config: ITestContainerConfig = {
+			registry: [
+				[mapId, SharedMap.getFactory()],
+				[cellId, SharedCell.getFactory()],
+			],
+			fluidDataObjectType: DataObjectFactoryType.Test,
+		};
+		const container1 = await provider.makeTestContainer(config);
+		const dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+		const map = await dataObject1.getSharedObject<SharedMap>(mapId);
+
+		assert(getIdCompressor(map) === undefined);
+	});
+
+	it("can normalize session space IDs to op space", async () => {
+		assert(getIdCompressor(sharedMap1) !== undefined);
+		assert(getIdCompressor(sharedMap2) !== undefined);
+		assert(getIdCompressor(sharedMap3) !== undefined);
+
+		const compressedIds: Map<SessionSpaceCompressedId, SharedMap> = new Map();
+		for (let i = 0; i < 1024; i++) {
+			compressedIds.set(getIdCompressor(sharedMap1).generateCompressedId(), sharedMap1);
+			compressedIds.set(getIdCompressor(sharedMap2).generateCompressedId(), sharedMap1);
```

Comment:
This generates the compressed id from container1's idCompressor but uses container2's idCompressor later to do notemalizeToOpSpace. Is that expected?

Reply:
This comment made me inspect the tests and do a pretty broad cleanup. Some tests were completely incorrect and were passing because my checks were unspecific (< 0, etc.) instead of asserting that the Id is actually the correct value. I've updated all the test here to check for the specific Ids that should be generated in all cases.